backend/src/modules/user/mutation.js

```diff
@@ -133,3 +134,57 @@ export const resetUserPassword = async (_, { email, newPassword }, { dbConnectio
     await dbConnection.query(`SELECT * FROM user WHERE email = ?`, [email])
   )[0];
 };
+
+export const setResetCode = async (
+  _,
+  { email },
+  { dbConnection },
+) => {
+    const code = random(99999999)
+    console.log("Code: ", code)
+
+    const setResetCodeDbResponse = await dbConnection.query(
+      `INSERT INTO forgotten (user_email, code)
+      VALUES (?, ?)`,
+      [email, code],
+    );
+
+    console.log("Insert ID: ", setResetCodeDbResponse.insertId)
+
+    if (setResetCodeDbResponse.insertId) {
+
+      console.log("I got here");
+      const link = "http://dev.frontend.team07.vse.handson.pro/password_reset/?email="
+        + email + "&code=" + code
+
+      console.log("Link: ", link)
+
+      const sgMail = require('@sendgrid/mail');
+      sgMail.setApiKey(process.env.SENDGRID_API_KEY);
+
+      const msg = {
+        to: email,
+        from: 'tym7nahlasto@gmail.com', // Nemenit!
+        subject: 'Change password confirmation',
+        text: 'You have successfully changed your password',
+        html: '<strong>'+ link + '</strong>',
+      };
+      sgMail
+        .send(msg)
+        .then(() => {
+          console.log('Email sent');
+        })
+        .catch((error) => {
+          //Log friendly error
+          console.log("Sendgrid failed.");
+          console.error(error.toString());
+          console.log(error.toString());
+        });
+    }
+
```

Comment:
Na toto prosím použij funkci z main. Doporučoval bych si mergnout main do této větve